src/pal/src/file/file.cpp

```diff
@@ -1250,8 +1287,17 @@ DeleteFileW(
       lpFileName?lpFileName:W16_NULLSTRING);
 
     pThread = InternalGetCurrentThread();
-    size = WideCharToMultiByte( CP_ACP, 0, lpFileName, -1, name, MAX_LONGPATH,
+    
+    if (lpFileName != NULL)
+    {
+        length = (PAL_wcslen(lpFileName)+1) * sizeof(WCHAR);
```

Comment:
It would be good to extract the length getting into a separate function, say `int GetMaxAcpStringSize(WCHAR* str)`